### PR TITLE
Feature/featured posts

### PR DESCRIPTION
**Pull request by iruzevic**
Added:
- featured posts
- featured categories
- featured categories posts

Storybook mock is not working because I need to find a way to mock API response from the block-render response for the ServerSideRender component

### CODE REVIEW

File: blocks/init/src/Blocks/custom/featured-categories-posts/docs/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Featured Categories Posts Block
+
+Block used to provide an Featured Posts functionality. You can simply copy/paste this block to your project and rename it to your new block needs.
```
Comment (dingo-d) [inline blocks/init/src/Blocks/custom/featured-categories-posts/docs/readme.md:3]:
```suggestion
Block used to provide featured posts functionality. You can simply copy/paste this block to your project and rename parts of it to fit your project (text domain mostly).
```

File: blocks/init/src/Blocks/custom/featured-categories-posts/featured-categories-posts.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+					],
+				];
+			}
+		};
+
+		if ($excludeCurrentPost) {
+			$args['post__not_in'] = [ $post->ID ];
+		}
+
+		$theQuery = new \WP_Query($args);
```
Comment (dingo-d) [inline blocks/init/src/Blocks/custom/featured-categories-posts/featured-categories-posts.php:56]: I'm not thrilled by the taxonomy queries or the `post__not_in`, as these are generally slow queries.

But we can profile later on to see if this is a bottleneck and come up with a solution later on.

File: blocks/init/src/Blocks/custom/featured-categories/docs/readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Featured Categories Block
+
+Block used to provide an Featured Posts functionality. You can simply copy/paste this block to your project and rename it to your new block needs.
```
Comment (dingo-d) [inline blocks/init/src/Blocks/custom/featured-categories/docs/readme.md:3]: Like the above readme comment, you can just copy/paste it.

File: blocks/init/src/Blocks/custom/featured-posts/docs/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Featured Posts Block
+
+Block used to provide an Featured Posts functionality. You can simply copy/paste this block to your project and rename it to your new block needs.
```
Comment (dingo-d) [inline blocks/init/src/Blocks/custom/featured-posts/docs/readme.md:3]: Same as the above

**Review by dingo-d — CHANGES_REQUESTED** [review]
Just minor readme text changes

**Review by IvanGrginovInf — APPROVED** [review]

**Review by dingo-d — APPROVED** [review]

**Pull request merged**